(MODULES-5576) Fix: iis_site Preload Enabled Not Working

ferventcoder opened the pull request:
Previously, the preload enabled setting for IIS Site was setting `serviceAutostartEnabled`
instead of preload enabled. Ensure that the setting sets 'applicationDefaults.preloadEnabled'
instead.

ferventcoder committed:
(MODULES-5576) Fix: iis_site preloadenabled not working

Previously, the preload enabled setting for IIS Site was setting service autostart enabled
instead of preload enabled. Ensure that the setting sets 'applicationDefaults.preloadEnabled'
instead.

jpogran commented:
You need to change the PR title one more time for this to be merged in ;)

jpogran commented:
I'll update the ticket to add an acceptance test for this. If you get to it before I do, I appreciate it

ferventcoder commented:
I think this is set so that you can add more to the PR. Go for it on the acceptance test. I assumed since it didn't fail on the change... ;)

ferventcoder commented:
yay, tests!

jpogran committed:
(MODULES-5576) Implement preloadenabled

This commit implements the rest of the code needed to make
preloadenabled work.

jpogran committed:
(MODULES-5576) Add test for preloadenabled

This commit adds an acceptance test for preloadenabled

jpogran commented:
this is ready for review

glennsarti commented on lib/puppet/provider/templates/webadministration/_getwebsites.ps1.erb:
```diff
@@ -22,5 +22,6 @@ Get-WebSite | %{
     logtruncatesize      = [string]$_.LogFile.truncateSize
     loglocaltimerollover = [string]$_.LogFile.localTimeRollover
     logextfileflags      = [string]$_.LogFile.logExtFileFlags
+    preloadenabled       = [string](Get-ItemProperty -Path "IIS:\Sites\$($_.Name)" -Name 'applicationDefaults.preloadEnabled' -ErrorAction 'Continue').Value
```
So this looks like the preloadEnabled element should really be defined on the application not at the site level.  Am I missing something here?

glennsarti replied:
Also this is setting the default of all applications within this site, whereas shouldn't we be modifying the specific application?

glennsarti replied:
Although I can see this gives us parity with the DSC implementation.

jpogran replied:
There are two different settings, application and site. I'm not understanding the comment here

glennsarti replied:
I guess my point was.  This is setting the application defaults, but there's corresponding parameter on the iis_application type for preloadenabled.  I guess this is ok so at least the bug is fixed as opposed to a new parameter too.

glennsarti commented on lib/puppet/provider/templates/webadministration/generalproperties.ps1.erb:
```diff
@@ -61,7 +61,7 @@ Try-SetItemProperty @logParams
 $logParams = @{
   Path  = "IIS:\Sites\$($resource.name)"
-  Name  = 'applicationDefaults.serviceAutoStartEnabled'
+  Name  = 'applicationDefaults.preloadEnabled'
```
👍

glennsarti reviewed (commented):
I think this is ok if we're looking for DSC parity

ferventcoder commented:
I'd click merge but am unable to now ;)
